{"url":"https://api.github.com/repos/elastic/elasticsearch/issues/31663","repository_url":"https://api.github.com/repos/elastic/elasticsearch","labels_url":"https://api.github.com/repos/elastic/elasticsearch/issues/31663/labels{/name}","comments_url":"https://api.github.com/repos/elastic/elasticsearch/issues/31663/comments","events_url":"https://api.github.com/repos/elastic/elasticsearch/issues/31663/events","html_url":"https://github.com/elastic/elasticsearch/issues/31663","id":336672744,"node_id":"MDU6SXNzdWUzMzY2NzI3NDQ=","number":31663,"title":"org.elasticsearch.transport.NodeNotConnectedException","user":{"login":"mohammedESSABRI","id":24996242,"node_id":"MDQ6VXNlcjI0OTk2MjQy","avatar_url":"https://avatars0.githubusercontent.com/u/24996242?v=4","gravatar_id":"","url":"https://api.github.com/users/mohammedESSABRI","html_url":"https://github.com/mohammedESSABRI","followers_url":"https://api.github.com/users/mohammedESSABRI/followers","following_url":"https://api.github.com/users/mohammedESSABRI/following{/other_user}","gists_url":"https://api.github.com/users/mohammedESSABRI/gists{/gist_id}","starred_url":"https://api.github.com/users/mohammedESSABRI/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/mohammedESSABRI/subscriptions","organizations_url":"https://api.github.com/users/mohammedESSABRI/orgs","repos_url":"https://api.github.com/users/mohammedESSABRI/repos","events_url":"https://api.github.com/users/mohammedESSABRI/events{/privacy}","received_events_url":"https://api.github.com/users/mohammedESSABRI/received_events","type":"User","site_admin":false},"labels":[],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":1,"created_at":"2018-06-28T15:32:59Z","updated_at":"2018-06-28T15:40:02Z","closed_at":"2018-06-28T15:40:02Z","author_association":"NONE","active_lock_reason":null,"body":"Hello team,\r\ni would like to ask a question about a bug which is appearing in Elasticseach since having a disk space issue, that was resolved but the bug still appearing\r\n<!--\r\n\r\n** Please read the guidelines below. **\r\n\r\nIssues that do not follow these guidelines are likely to be closed.\r\n\r\n1.  GitHub is reserved for bug reports and feature requests. The best place to\r\n    ask a general question is at the Elastic [forums](https://discuss.elastic.co).\r\n    GitHub is not the place for general questions.\r\n\r\n2.  Is this bug report or feature request for a supported OS? If not, it\r\n    is likely to be closed.  See https://www.elastic.co/support/matrix#show_os\r\n\r\n3.  Please fill out EITHER the feature request block or the bug report block\r\n    below, and delete the other block.\r\n\r\n-->\r\n\r\n<!-- Bug report -->\r\n\r\n**Elasticsearch version** (`bin/elasticsearch --version`):\r\n\r\n1.7\r\n\r\n**Plugins installed**: []\r\n\r\n**JVM version** (`java -version`):\r\n7.0.1\r\n**OS version** (`uname -a` if on a Unix-like system):\r\nLinux 4.482-6.9-default x86_64\r\n\r\n**Description of the problem including expected versus actual behavior**:\r\n* We have 6 environments: local, test, env3, env4, env5 and prod.\r\n* in each environment we have Elasticsearch server and Java EE application which communicating with Elasticsearch to provide search services.\r\n* in each environment, we have two nodes for Java EE app and two nodes for Elasticsearch server\r\n\r\nThe problem is that we had errors in search, then the server is not working correctly, please see steps below.\r\n\r\n\r\n**Steps to reproduce**:\r\n\r\n* Before, everything was working good.\r\n\r\n* After that, a user of Java EE application did a test in env4 and get an error in search ( caused by a problem in language filter ) \r\n\r\n* Trying to resolve the problem we compared it with prod ( same JEE app version, same ES version and same index mapping, and also same database db2), the result is that it works on PROD.\r\n\r\n* We deployed the same JEE app and same mapping, same ES version in test ( the same as in env4 ) and it's working\r\n\r\n* We did the same in env3 and it's not working\r\n\r\n* We pointed local env to Elasticsearch server and database from test env , and we used the same version of JEE app,  and we tested but it's not working\r\n\r\n* We get the query that the JEE app sends to Elasticsearch and we compared the two cases : when we have the problem and when we don't get the problem, so we arrived to the following Elasticsearch request, we tested it using curl but i didn't give any hits in result:\r\n\r\nNote : http://localhost is the ip of our server, i didn't share it for security reasons.\r\n\r\n```\r\ncurl -XGET 'http://localhost/_search?pretty' -d'{\r\n  \"from\" : 0,\r\n  \"size\" : 6,\r\n  \"query\" : {\r\n    \"filtered\" : {\r\n      \"query\" : {\r\n        \"bool\" : {\r\n          \"must\" : [ {\r\n            \"bool\" : {\r\n              \"must\" : {\r\n                \"multi_match\" : {\r\n                  \"query\" : \"EN\",\r\n                  \"fields\" : [ \"proceedingslanguage_*\", \"proceedingslanguage_en^5.0\" ],\r\n                  \"type\" : \"best_fields\"\r\n                }\r\n              }\r\n            }\r\n          } ]\r\n        }\r\n      },\r\n      \"filter\" : {\r\n        \"type\" : {\r\n          \"value\" : \"decision\"\r\n        }\r\n      }\r\n    }\r\n  }\r\n}'\r\n```\r\n\r\nThe result:\r\n\r\n```\r\n   {\r\n      \"took\" : 10,\r\n      \"timed_out\" : false,\r\n      \"_shards\" : {\r\n        \"total\" : 27,\r\n        \"successful\" : 27,\r\n        \"failed\" : 0\r\n      },\r\n      \"hits\" : {\r\n        \"total\" : 0,\r\n        \"max_score\" : null,\r\n        \"hits\" : [ ]\r\n      }\r\n    }\r\n\r\n```\r\n\r\n\r\n* After we get error in log: 85% of space disk is full in env4 and env3:\r\n\r\nExtract of log in  node1:\r\n> [2018-06-26 16:29:39,081][INFO ][cluster.routing.allocation.decider] [Elasticsearch_serverName_node2] low disk watermark [85%] exceeded on [-5BLsRz5RTas....][Elasticsearch_serverName_node2] free: 1gb[12.1%], replicas will not be assigned to this node\r\n\r\n* After that we get error authentification in env4\r\n\r\n> [2018-06-26 18:36:46,952][WARN ][shield.authc.activedirectory] [Elasticsearch_serverName_node1] \r\n> authentication failed for user [userName]: unable to authenticate user [userName] to active directory domain [domaineName]       cause: com.unboundid.ldap.sdk.LDAPException: 80090308: LdapErr: DSID-0C090400, comment: AcceptSecurityContext error...\r\n\r\n* After we get error  HTTP interface http://hostName:9300 is not connected\r\n\r\n> [2018-06-27 17:38:55,370][WARN ][discovery.zen.ping.unicast] [Elasticsearch_serverName_node1] failed to send ping to [[#zen_unicast_1#][Elasticsearch_serverName_node1][inet[/hostName:9300]]]\r\n> org.elasticsearch.transport.SendRequestTransportException: [Elasticsearch_serverName_node1][inet[/hostName:9300]][internal:discovery/zen/unicast]\r\n> \tat org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:286)\r\n> \tat org.elasticsearch.shield.transport.ShieldServerTransportService.sendRequest(ShieldServerTransportService.java:59)\r\n> \tat org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.sendPingRequestToNode(UnicastZenPing.java:468)\r\n> \tat org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.access$1000(UnicastZenPing.java:62)\r\n> \tat org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing$3.run(UnicastZenPing.java:383)\r\n> \tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)\r\n> \tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)\r\n> \tat java.lang.Thread.run(Thread.java:745)\r\n> Caused by: org.elasticsearch.transport.NodeNotConnectedException: [Elasticsearch_serverName_node1][inet[/hostName:9300]] Node not connected\r\n> \tat org.elasticsearch.transport.netty.NettyTransport.nodeChannel(NettyTransport.java:964)\r\n> \tat org.elasticsearch.transport.netty.NettyTransport.sendRequest(NettyTransport.java:656)\r\n> \tat org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:276)\r\n> \t... 7 more\r\n\r\n* we asked responsible team to extend space in disk so the error relative to disk page was resolved.\r\n* Even after the step above, the problem of \"Node not connected exception\" continue appearing in log file\r\n\r\n\r\nPlease include a *minimal* but *complete* recreation of the problem, including\r\n(e.g.) index creation, mappings, settings, query etc.  The easier you make for\r\nus to reproduce it, the more likely that somebody will take the time to look at it.\r\n\r\n 1.\r\n 2.\r\n 3.\r\n\r\n\r\nThank you and best regards,\r\nMohammed ESSABRI\r\n\r\n","closed_by":{"login":"colings86","id":236731,"node_id":"MDQ6VXNlcjIzNjczMQ==","avatar_url":"https://avatars0.githubusercontent.com/u/236731?v=4","gravatar_id":"","url":"https://api.github.com/users/colings86","html_url":"https://github.com/colings86","followers_url":"https://api.github.com/users/colings86/followers","following_url":"https://api.github.com/users/colings86/following{/other_user}","gists_url":"https://api.github.com/users/colings86/gists{/gist_id}","starred_url":"https://api.github.com/users/colings86/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/colings86/subscriptions","organizations_url":"https://api.github.com/users/colings86/orgs","repos_url":"https://api.github.com/users/colings86/repos","events_url":"https://api.github.com/users/colings86/events{/privacy}","received_events_url":"https://api.github.com/users/colings86/received_events","type":"User","site_admin":false},"performed_via_github_app":null}